### DISCUSSION TITLE
Embeding a Vulkan surface

### DISCUSSION THREAD

**Discussion by ArthurVasseur** [AvaloniaUI/Avalonia#12896] [General]
I have a graphical engine written in C++ and Vulkan.
I would like to know if there is a way to draw with my engine in the gray rectangle ?

![image](https://github.com/AvaloniaUI/Avalonia/assets/72025980/6e40028c-c6a2-4347-9fd4-5fc15e33680f)

Maybe I can get a vulkan surface from Avalonia ?

**Comment by kekekeks** [comment]
See https://github.com/AvaloniaUI/Avalonia/issues/9925 and https://github.com/AvaloniaUI/Avalonia/tree/master/samples/GpuInterop

**Reply by ArthurVasseur** [reply]
Thank you for your response. However, I'm having trouble locating where you've assigned the `VkSurfaceKHR` in the code. I've tryed to search through the code, but unfortunately, I couldn't find it.

**Reply by kekekeks** [reply]
There is no VkSurfaceKHR. See the interop spec.
